### DISCUSSION TITLE
How to set $args value in proxy_rewrite plugin

### DISCUSSION THREAD

**Discussion by huang-x-h** [apache/apisix#7000] [Q&A]
Now, we want to change the $args value, append something

the nginx conf 
```
location /ApiService/ {
    set $args "$args&appKey=secretKey";
    proxy_pass https://somepath.example.com/;
}
```
when user access `http://ip:port/ApiService/getUser?userId=123`, it will be proxy to `https://somepath.example.com/getUser?userId=123&appKey=secretKey`

then we use `proxy_rewrite` plugin
```
{
  "regex_uri": [
    "^/ApiService/(.*)",
    "/$1&appKey=secretKey"
  ]
}
```
but it just rewrite uri, not include the parameters, so how to implement parameter rewriting? use `serverless` plugin?

**Comment by huang-x-h** [comment]
It's fixed, see the `proxy_rewrite` source code, it will append parameter, so we change the conf

```
{
  "regex_uri": [
    "^/ApiService/(.*)",
    "/$1?appKey=secretKey"
  ]
}
```

**Reply by showcup** [reply]
yes,I was looking right here, too
<img width="548" alt="image" src="https://user-images.githubusercontent.com/319657/211341318-6b410e03-76c4-4542-89e9-222b55de29ac.png">
